ci-operator/config/redhat-appstudio/jvm-build-service/redhat-appstudio-jvm-build-service-main.yaml

```diff
@@ -6,9 +6,11 @@ build_root:
 images:
 - dockerfile_path: Dockerfile
   to: redhat-appstudio-jvm-build-service-image
-- dockerfile_path: Dockerfile.cache-in-ci
+- context_dir: java-components
+  dockerfile_path: java-components/cache/src/main/docker/Dockerfile.all-in-one
```

Comment:
```suggestion
  dockerfile_path: cache/src/main/docker/Dockerfile.all-in-one
```

Reply:
I *think* 

and that matches what you have for build-requests below

and as I read follows a pattern I've seen work elsewhere, for example
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/jenkins/openshift-jenkins-master.yaml#L37-L44

and 


https://github.com/openshift/jenkins/tree/master/agent-maven

what I'm unclear on is if you'll need the 
```
inputs:
   src:
       paths:
      
...
```

I'll see if I can sort that out, and report back

Reply:
trying some stuff out in https://github.com/openshift/release/pull/38006 ... if some of that doesn't work, or if my suggestion here does not, we may be stuck

and the more I try to recollect, more I think the dichotomy here between how the `mvn` stuff works with how openshift ci, with its primary focus on golang based projects, may have been why I just created the duplicate docker files in the first place

but #38006 is still running as I type

Reply:
or maybe this is right, but build-request-processor needs to be updated ? ... trying that over in #38006 in a couple

Reply:
yep confirmed in https://github.com/openshift/release/pull/38006 with https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/38006/rehearse-38006-pull-ci-redhat-appstudio-jvm-build-service-main-images/1642982614287519744 

My original suggestion was in fact correct.  No need for the `inputs` stuff ...

Reply:
oops, I missed that I forgot to update that path.